apps/fxc-front/src/app/logic/score/league/leagues.ts

```diff
@@ -1,22 +1,19 @@
-import { League } from '../league';
-import { CzechEurope, CzechLocal, CzechOutEurope } from './czech';
-import { FrCfd } from './frcfd';
-import { Leonardo } from './leonardo';
-import { UKXCLClub, UKXCLInternational, UKXCLNational } from './ukxcl';
-import { WXC } from './wxc';
-import { NorwayLeague, XContest, XContestPPG } from './xcontest';
-
-export const LEAGUES: { [name: string]: League } = {
-  czl: new CzechLocal(),
-  cze: new CzechEurope(),
-  czo: new CzechOutEurope(),
-  fr: new FrCfd(),
-  leo: new Leonardo(),
-  nor: new NorwayLeague(),
-  ukc: new UKXCLClub(),
-  uki: new UKXCLInternational(),
-  ukn: new UKXCLNational(),
-  xc: new XContest(),
-  xcppg: new XContestPPG(),
-  wxc: new WXC(),
+export const LEAGUES: { [name: string]: string } = {
+  czl: 'Czech (ČPP local)',
+  cze: 'Czech (ČPP Europe)',
+  czo: 'Czech (ČPP outside Europe)',
+  fr: 'France (CFD)',
+  leo: 'Leonardo',
+  nor: 'Norway (Distanseligaen)',
+  ukc: 'UK (XC League, Club)',
+  uki: 'UK (XC League, International)',
+  ukn: 'UK (XC League, National)',
+  xc: 'XContest',
+  xcppg: 'XContest PPG',
+  wxc: 'World XC Online Contest',
 };
+
+// allowed league codes
```

Comment:
A better way to do this is:

```typescript
export const trackerNames = ['inreach', 'spot', 'skylines', 'flyme', 'flymaster', 'ogn', 'zoleo', 'xcontest'] as const;

// ID for the tracking devices.
export type TrackerNames = (typeof trackerNames)[number];

// How to display the tracker name.
export const trackerDisplayNames: Readonly<Record<TrackerNames, string>> = {
  inreach: 'InReach',
  spot: 'Spot',
  skylines: 'Skylines',
  flyme: 'FlyMe (XCGlobe)',
  flymaster: 'Flymaster',
  ogn: 'OGN',
  zoleo: 'zoleo',
  xcontest: 'XContest',
};
```

Then we can also iterate on the leagues

Reply:
code changed. 
...But it's a bit tricky :-/ As I am not at all a TS/JS expert, I struggle with such details

Reply:
I struggled too and had to do a good fair of googling before finding this trick

Reply:
Surprisingly, TS enums implementation is quite poor comparing with other languages like Kotlin/Java/Scala